utils/wikize_refs.py

```diff
@@ -5,44 +5,49 @@
+    from urllib2 import urlopen
+    from urllib2 import Request
+except:
+    # python3
+    from urllib.parse import urlparse
+    from urllib.request import urlopen
+    from urllib.request import Request
+
 def usage():
     return \
 """
-Re-formats a series of reference style links in a GitHub Markdown file
-so that the article's footnote links behave more Wikipedia-like.
-
-Treats a markdown file as being composed of four successive logical
-blocks...
-
-  1. The main content with possible footnote refs (<sup>[J],...</sup>)
-  2. The author's link definitions ([J]: https://... "Title {}")
-  3. Auto-gen'd Intermediate link definitions ([J]: #refJ "Title {}")
-  4. Auto-gen'd Ref. table (<a name="refJ"></a>J | [Title](https://... "{...}"))
-
-Blocks 2, 3 and 4 are optional. Blocks 3 and 4 are generated from
-block 2 if it exists. Repeated applications of this tool should
-result in no changes to the file.
-
-If the main content contains no footnotes, the file will be left
-unchanged. If a footnote references a non-existent author's link
-definition, an error message is issued and processing is stopped.
-If a link definition is not referenced by any footnote, a warning
-message is produced but processing continues.
-
-The author's link definitions are renumbered 1...N and all footnote
-references are updated accordingly. These link definitions are output
-but bracketed within XML comment blocks to hide these link definitions
-from any markdown processing.
-
-The renumbered and re-formatted links are bi-level. Footnotes in the
-content link to entries in a list of references at the bottom of the
-document. Items in the table of references link off-page to their
+Adjusts, slightly, a markdown file so that any footnotes using
+reference-style markdown links and link-definitions behave more
+Wikipedia-like. Ordinarily, this can be accomplished by simply
+name-mangling the author's original link labels (to move them
+out of the way) and *adding* some lines to the file.
+
+The author's *original* link labels get mangled. This has the effect
+of breaking their connection to the footnotes referencing them. This
+is later repaired when a new set of link definitions are appended
+to the end of the file.
+
+The new links are bi-level. Footnotes in the content link to
+entries in a visible list of references appended to the end of the
+document. Items in the list of references link off-page to their
 intended destinations. The resulting file is still GitHub flavored
 Markdown with a minimal amount of embedded HTML.
 
-For more information, see https://github.com/betterscientificsoftware/\
-betterscientificsoftware.github.io/blob/master/Articles/Blog/\
-ReferencesInMarkdownHybridApproach.md
+If the file contains no footnotes, it will be left unchanged.
+Some minimal error checks are that there is an existing link
+definition for every footnote and that every link definition appears
+in at least one footnote. It will also check that links are valid.
```

Comment:
Not sure what this sentence is saying:

> Some minimal error checks are that there is an existing link definition for every footnote and that every link definition appears in at least one footnote.

Would this be more clear as:

> Some minimal error checks include: a) there is an existing link definition for every footnote and b) every link definition appears in at least one footnote.

?

Also, what is meant by:

> It will also check that links are valid.

?

Does that mean it will do network communication and does it by default?  I think the answer is "yes" and "no" but that needs to be made more clear.  Also, the option `-c CHECK_LINKS, --check-links CHECK_LINKS` needs to make it clear that this will actually do **network communication** (which one would not ordinarily expect for a simple utility)!

Reply:
Done